netket/exact.py

```diff
@@ -25,3 +65,132 @@ def _ImagTimePropagation_iter(self, dt, n_iter=None):
 
 
 ImagTimePropagation.iter = _ImagTimePropagation_iter
+
+
+@deprecated()
+class EdResult(object):
+    def __init__(self, eigenvalues, eigenvectors):
+        # NOTE: These conversions are required because our old C++ code stored
+        # eigenvalues and eigenvectors as Python lists :(
+        self._eigenvalues = eigenvalues.tolist()
+        self._eigenvectors = (
+            [eigenvectors[:, i] for i in range(eigenvectors.shape[1])]
+            if eigenvectors is not None
+            else []
+        )
+
+    @property
+    def eigenvalues(self):
+        r"""Eigenvalues of the Hamiltonian.
+        """
+        return self._eigenvalues
+
+    @property
+    def eigenvectors(self):
+        r"""Eigenvectors of the Hamiltonian.
+        """
+        return self._eigenvectors
+
+    def mean(self, operator, which):
+        import numpy
+
+        x = self._eigenvectors[which]
+        return numpy.vdot(x, operator(x))
+
+
+@deprecated(
+    "use 'netket.Operator.to_linear_operator' to convert an operator "
+    "to a 'scipy.sparse.linalg.LinearOperator' and then call "
+    "'scipy.sparse.linalg.eigsh' directly"
+)
+def lanczos_ed(
+    operator,
+    matrix_free=False,
+    first_n=1,
+    max_iter=1000,
+    seed=None,
+    precision=1e-14,
+    compute_eigenvectors=False,
+):
+    r"""Uses Lanczos algorithm to diagonalize the operator.
+
+    Args:
+        operator: The operator to diagnolize.
+        matrix_free: If true, matrix elements are computed on the fly.
+            Otherwise, the operator is first converted to a sparse matrix.
+        first_n: The number of eigenvalues to compute.
+        max_iter: The maximum number of iterations.
+        seed: **Ignored**. Accepted for backward compatibility only.
+        precision: The precision to which the eigenvalues will be
+            computed.
+        compute_eigenvectors: Whether or not to compute the
```

Comment:
we should comment on the fact that this option does not have any influence on the actual computation of the eigenvectors, when using the scipy backend